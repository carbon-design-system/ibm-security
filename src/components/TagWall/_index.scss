////
/// Tag wall.
/// @group tag-wall
/// @copyright IBM Security 2019
////

@import '../Component/mixins';

@import 'mixins';

@import '../Tag/index';

@include security--component($name: tag-wall) {
  @include tag-wall;
}
