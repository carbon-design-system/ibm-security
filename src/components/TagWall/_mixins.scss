////
/// Tag wall mixins.
/// @group tag-wall
/// @copyright IBM Security 2019 - 2021
////

@import '@carbon/type/scss/index';

@mixin tag-wall {
  &__label {
    $tag-wall__spacing: $carbon--spacing-03;

    @include carbon--type-style($name: caption-01);
    margin-top: $tag-wall__spacing;
    margin-bottom: $tag-wall__spacing;
  }

  > #{$tag--interactive__namespace} {
    max-width: 40ex;

    &__button {
      flex-shrink: 0;
    }
  }
}
