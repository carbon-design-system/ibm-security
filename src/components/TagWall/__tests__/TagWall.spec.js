/**
 * @file Tag wall tests.
 * @copyright IBM Security 2019
 */

import { mount } from 'enzyme';
import React from 'react';

import { Button, InteractiveTag, TagWall } from '../../..';

import { type } from '../TagWall';
import { namespace as interactiveTagNamespace } from '../../Tag/InteractiveTag/InteractiveTag';

import props from '../_mocks_';

describe('TagWall', () => {
  let tagWall;

  beforeEach(() => {
    tagWall = mount(<TagWall {...props} />);
  });

  describe('Rendering', () => {
    it('renders', () => {
      expect(tagWall).toMatchSnapshot();
    });

    it("renders the HTML of the node's subtree", () => {
      expect(tagWall.render()).toMatchSnapshot();
    });

    it('renders the disabled variation', () => {
      tagWall.setProps({ disable: true });

      expect(tagWall).toMatchSnapshot();
    });
  });

  describe('Events', () => {
    const { fn } = jest;

    // Get the "close" button:
    const getButton = () =>
      tagWall
        .find(InteractiveTag)
        .first()
        .find(`button.${interactiveTagNamespace}__button`);

    it('should call the `onChange` method', () => {
      const onChange = fn();
      tagWall.setProps({ onChange });

      getButton().simulate('click');
      expect(onChange).toHaveBeenCalledWith(expect.objectContaining({ type }));
    });

    it('should stop propagation when the `onChange` method is called', () => {
      const stopPropagation = fn();

      getButton().simulate('click', { stopPropagation });
      expect(stopPropagation).toHaveBeenCalled();
    });

    it('should call the `onAddButton` method', () => {
      const onAddButton = fn();
      tagWall.setProps({ onAddButton });

      tagWall.find(Button).simulate('click');
      expect(onAddButton).toHaveBeenCalled();
    });
  });
});
