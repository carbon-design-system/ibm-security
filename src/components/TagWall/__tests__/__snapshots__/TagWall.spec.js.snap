// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagWall Rendering renders 1`] = `
<TagWall
  addButtonDisabled={false}
  addLabel="Add"
  className=""
  disable={false}
  itemToString={[Function]}
  items={
    Array [
      Object {
        "id": "0",
        "isSelected": true,
        "label": "Tag 1",
      },
      Object {
        "id": "1",
        "isSelected": true,
        "label": "Tag 2",
      },
      Object {
        "id": "2",
        "isSelected": true,
        "label": "Tag 3",
      },
      Object {
        "id": "3",
        "isSelected": false,
        "label": "Tag 4",
      },
      Object {
        "id": "4",
        "isSelected": false,
        "label": "Tag 5",
      },
      Object {
        "id": "5",
        "isSelected": false,
        "label": "Tag 6",
      },
      Object {
        "id": "6",
        "isSelected": false,
        "label": "Tag 7",
      },
      Object {
        "id": "7",
        "isSelected": false,
        "label": "Tag 8",
      },
    ]
  }
  label="Label"
  labels={Object {}}
  onAddButton={[Function]}
  onChange={[Function]}
>
  <section
    aria-label="Label"
    className="security--tag-wall"
  >
    <h2
      className="security--tag-wall__label"
    >
      Label
    </h2>
    <InteractiveTag
      isSelected={true}
      key="tag__0"
      onRemove={[Function]}
      removable={true}
      removeBtnLabel=""
      type="gray"
    >
      <Tag
        className="security--tag--interactive security--tag--interactive--removable security--tag--interactive--selected"
        filter={false}
        type="gray"
      >
        <span
          className="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--removable security--tag--interactive--selected"
        >
          Tag 1
          <button
            aria-label=""
            className="security--tag--interactive__button"
            onClick={[Function]}
          >
            <Icon
              className="security--tag--interactive__button__icon"
              height={null}
              path={null}
              renderIcon={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                }
              }
              size={20}
              viewBox="0 0 32 32"
              width={null}
            >
              <ForwardRef(Close16)
                aria-hidden="true"
                className="security--icon security--tag--interactive__button__icon"
                focusable={false}
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "willChange": "transform",
                  }
                }
              >
                <Icon
                  aria-hidden="true"
                  className="security--icon security--tag--interactive__button__icon"
                  focusable={false}
                  height={16}
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "willChange": "transform",
                    }
                  }
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <svg
                    aria-hidden={true}
                    className="security--icon security--tag--interactive__button__icon"
                    focusable={false}
                    height={16}
                    preserveAspectRatio="xMidYMid meet"
                    style={
                      Object {
                        "willChange": "transform",
                      }
                    }
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 4.7L11.3 4 8 7.3 4.7 4 4 4.7 7.3 8 4 11.3 4.7 12 8 8.7 11.3 12 12 11.3 8.7 8z"
                    />
                  </svg>
                </Icon>
              </ForwardRef(Close16)>
            </Icon>
          </button>
        </span>
      </Tag>
    </InteractiveTag>
    <InteractiveTag
      isSelected={true}
      key="tag__1"
      onRemove={[Function]}
      removable={true}
      removeBtnLabel=""
      type="gray"
    >
      <Tag
        className="security--tag--interactive security--tag--interactive--removable security--tag--interactive--selected"
        filter={false}
        type="gray"
      >
        <span
          className="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--removable security--tag--interactive--selected"
        >
          Tag 2
          <button
            aria-label=""
            className="security--tag--interactive__button"
            onClick={[Function]}
          >
            <Icon
              className="security--tag--interactive__button__icon"
              height={null}
              path={null}
              renderIcon={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                }
              }
              size={20}
              viewBox="0 0 32 32"
              width={null}
            >
              <ForwardRef(Close16)
                aria-hidden="true"
                className="security--icon security--tag--interactive__button__icon"
                focusable={false}
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "willChange": "transform",
                  }
                }
              >
                <Icon
                  aria-hidden="true"
                  className="security--icon security--tag--interactive__button__icon"
                  focusable={false}
                  height={16}
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "willChange": "transform",
                    }
                  }
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <svg
                    aria-hidden={true}
                    className="security--icon security--tag--interactive__button__icon"
                    focusable={false}
                    height={16}
                    preserveAspectRatio="xMidYMid meet"
                    style={
                      Object {
                        "willChange": "transform",
                      }
                    }
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 4.7L11.3 4 8 7.3 4.7 4 4 4.7 7.3 8 4 11.3 4.7 12 8 8.7 11.3 12 12 11.3 8.7 8z"
                    />
                  </svg>
                </Icon>
              </ForwardRef(Close16)>
            </Icon>
          </button>
        </span>
      </Tag>
    </InteractiveTag>
    <InteractiveTag
      isSelected={true}
      key="tag__2"
      onRemove={[Function]}
      removable={true}
      removeBtnLabel=""
      type="gray"
    >
      <Tag
        className="security--tag--interactive security--tag--interactive--removable security--tag--interactive--selected"
        filter={false}
        type="gray"
      >
        <span
          className="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--removable security--tag--interactive--selected"
        >
          Tag 3
          <button
            aria-label=""
            className="security--tag--interactive__button"
            onClick={[Function]}
          >
            <Icon
              className="security--tag--interactive__button__icon"
              height={null}
              path={null}
              renderIcon={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                }
              }
              size={20}
              viewBox="0 0 32 32"
              width={null}
            >
              <ForwardRef(Close16)
                aria-hidden="true"
                className="security--icon security--tag--interactive__button__icon"
                focusable={false}
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "willChange": "transform",
                  }
                }
              >
                <Icon
                  aria-hidden="true"
                  className="security--icon security--tag--interactive__button__icon"
                  focusable={false}
                  height={16}
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "willChange": "transform",
                    }
                  }
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <svg
                    aria-hidden={true}
                    className="security--icon security--tag--interactive__button__icon"
                    focusable={false}
                    height={16}
                    preserveAspectRatio="xMidYMid meet"
                    style={
                      Object {
                        "willChange": "transform",
                      }
                    }
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 4.7L11.3 4 8 7.3 4.7 4 4 4.7 7.3 8 4 11.3 4.7 12 8 8.7 11.3 12 12 11.3 8.7 8z"
                    />
                  </svg>
                </Icon>
              </ForwardRef(Close16)>
            </Icon>
          </button>
        </span>
      </Tag>
    </InteractiveTag>
    <InteractiveTag
      isSelected={false}
      key="tag__3"
      onRemove={[Function]}
      removable={true}
      removeBtnLabel=""
      type="gray"
    >
      <Tag
        className="security--tag--interactive security--tag--interactive--default security--tag--interactive--removable"
        filter={false}
        type="gray"
      >
        <span
          className="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--default security--tag--interactive--removable"
        >
          Tag 4
          <button
            aria-label=""
            className="security--tag--interactive__button"
            onClick={[Function]}
          >
            <Icon
              className="security--tag--interactive__button__icon"
              height={null}
              path={null}
              renderIcon={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                }
              }
              size={20}
              viewBox="0 0 32 32"
              width={null}
            >
              <ForwardRef(Close16)
                aria-hidden="true"
                className="security--icon security--tag--interactive__button__icon"
                focusable={false}
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "willChange": "transform",
                  }
                }
              >
                <Icon
                  aria-hidden="true"
                  className="security--icon security--tag--interactive__button__icon"
                  focusable={false}
                  height={16}
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "willChange": "transform",
                    }
                  }
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <svg
                    aria-hidden={true}
                    className="security--icon security--tag--interactive__button__icon"
                    focusable={false}
                    height={16}
                    preserveAspectRatio="xMidYMid meet"
                    style={
                      Object {
                        "willChange": "transform",
                      }
                    }
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 4.7L11.3 4 8 7.3 4.7 4 4 4.7 7.3 8 4 11.3 4.7 12 8 8.7 11.3 12 12 11.3 8.7 8z"
                    />
                  </svg>
                </Icon>
              </ForwardRef(Close16)>
            </Icon>
          </button>
        </span>
      </Tag>
    </InteractiveTag>
    <InteractiveTag
      isSelected={false}
      key="tag__4"
      onRemove={[Function]}
      removable={true}
      removeBtnLabel=""
      type="gray"
    >
      <Tag
        className="security--tag--interactive security--tag--interactive--default security--tag--interactive--removable"
        filter={false}
        type="gray"
      >
        <span
          className="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--default security--tag--interactive--removable"
        >
          Tag 5
          <button
            aria-label=""
            className="security--tag--interactive__button"
            onClick={[Function]}
          >
            <Icon
              className="security--tag--interactive__button__icon"
              height={null}
              path={null}
              renderIcon={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                }
              }
              size={20}
              viewBox="0 0 32 32"
              width={null}
            >
              <ForwardRef(Close16)
                aria-hidden="true"
                className="security--icon security--tag--interactive__button__icon"
                focusable={false}
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "willChange": "transform",
                  }
                }
              >
                <Icon
                  aria-hidden="true"
                  className="security--icon security--tag--interactive__button__icon"
                  focusable={false}
                  height={16}
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "willChange": "transform",
                    }
                  }
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <svg
                    aria-hidden={true}
                    className="security--icon security--tag--interactive__button__icon"
                    focusable={false}
                    height={16}
                    preserveAspectRatio="xMidYMid meet"
                    style={
                      Object {
                        "willChange": "transform",
                      }
                    }
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 4.7L11.3 4 8 7.3 4.7 4 4 4.7 7.3 8 4 11.3 4.7 12 8 8.7 11.3 12 12 11.3 8.7 8z"
                    />
                  </svg>
                </Icon>
              </ForwardRef(Close16)>
            </Icon>
          </button>
        </span>
      </Tag>
    </InteractiveTag>
    <InteractiveTag
      isSelected={false}
      key="tag__5"
      onRemove={[Function]}
      removable={true}
      removeBtnLabel=""
      type="gray"
    >
      <Tag
        className="security--tag--interactive security--tag--interactive--default security--tag--interactive--removable"
        filter={false}
        type="gray"
      >
        <span
          className="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--default security--tag--interactive--removable"
        >
          Tag 6
          <button
            aria-label=""
            className="security--tag--interactive__button"
            onClick={[Function]}
          >
            <Icon
              className="security--tag--interactive__button__icon"
              height={null}
              path={null}
              renderIcon={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                }
              }
              size={20}
              viewBox="0 0 32 32"
              width={null}
            >
              <ForwardRef(Close16)
                aria-hidden="true"
                className="security--icon security--tag--interactive__button__icon"
                focusable={false}
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "willChange": "transform",
                  }
                }
              >
                <Icon
                  aria-hidden="true"
                  className="security--icon security--tag--interactive__button__icon"
                  focusable={false}
                  height={16}
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "willChange": "transform",
                    }
                  }
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <svg
                    aria-hidden={true}
                    className="security--icon security--tag--interactive__button__icon"
                    focusable={false}
                    height={16}
                    preserveAspectRatio="xMidYMid meet"
                    style={
                      Object {
                        "willChange": "transform",
                      }
                    }
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 4.7L11.3 4 8 7.3 4.7 4 4 4.7 7.3 8 4 11.3 4.7 12 8 8.7 11.3 12 12 11.3 8.7 8z"
                    />
                  </svg>
                </Icon>
              </ForwardRef(Close16)>
            </Icon>
          </button>
        </span>
      </Tag>
    </InteractiveTag>
    <InteractiveTag
      isSelected={false}
      key="tag__6"
      onRemove={[Function]}
      removable={true}
      removeBtnLabel=""
      type="gray"
    >
      <Tag
        className="security--tag--interactive security--tag--interactive--default security--tag--interactive--removable"
        filter={false}
        type="gray"
      >
        <span
          className="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--default security--tag--interactive--removable"
        >
          Tag 7
          <button
            aria-label=""
            className="security--tag--interactive__button"
            onClick={[Function]}
          >
            <Icon
              className="security--tag--interactive__button__icon"
              height={null}
              path={null}
              renderIcon={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                }
              }
              size={20}
              viewBox="0 0 32 32"
              width={null}
            >
              <ForwardRef(Close16)
                aria-hidden="true"
                className="security--icon security--tag--interactive__button__icon"
                focusable={false}
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "willChange": "transform",
                  }
                }
              >
                <Icon
                  aria-hidden="true"
                  className="security--icon security--tag--interactive__button__icon"
                  focusable={false}
                  height={16}
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "willChange": "transform",
                    }
                  }
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <svg
                    aria-hidden={true}
                    className="security--icon security--tag--interactive__button__icon"
                    focusable={false}
                    height={16}
                    preserveAspectRatio="xMidYMid meet"
                    style={
                      Object {
                        "willChange": "transform",
                      }
                    }
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 4.7L11.3 4 8 7.3 4.7 4 4 4.7 7.3 8 4 11.3 4.7 12 8 8.7 11.3 12 12 11.3 8.7 8z"
                    />
                  </svg>
                </Icon>
              </ForwardRef(Close16)>
            </Icon>
          </button>
        </span>
      </Tag>
    </InteractiveTag>
    <InteractiveTag
      isSelected={false}
      key="tag__7"
      onRemove={[Function]}
      removable={true}
      removeBtnLabel=""
      type="gray"
    >
      <Tag
        className="security--tag--interactive security--tag--interactive--default security--tag--interactive--removable"
        filter={false}
        type="gray"
      >
        <span
          className="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--default security--tag--interactive--removable"
        >
          Tag 8
          <button
            aria-label=""
            className="security--tag--interactive__button"
            onClick={[Function]}
          >
            <Icon
              className="security--tag--interactive__button__icon"
              height={null}
              path={null}
              renderIcon={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                }
              }
              size={20}
              viewBox="0 0 32 32"
              width={null}
            >
              <ForwardRef(Close16)
                aria-hidden="true"
                className="security--icon security--tag--interactive__button__icon"
                focusable={false}
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "willChange": "transform",
                  }
                }
              >
                <Icon
                  aria-hidden="true"
                  className="security--icon security--tag--interactive__button__icon"
                  focusable={false}
                  height={16}
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "willChange": "transform",
                    }
                  }
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <svg
                    aria-hidden={true}
                    className="security--icon security--tag--interactive__button__icon"
                    focusable={false}
                    height={16}
                    preserveAspectRatio="xMidYMid meet"
                    style={
                      Object {
                        "willChange": "transform",
                      }
                    }
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 4.7L11.3 4 8 7.3 4.7 4 4 4.7 7.3 8 4 11.3 4.7 12 8 8.7 11.3 12 12 11.3 8.7 8z"
                    />
                  </svg>
                </Icon>
              </ForwardRef(Close16)>
            </Icon>
          </button>
        </span>
      </Tag>
    </InteractiveTag>
    <Button
      disabled={false}
      kind="ghost"
      largeText={null}
      loading={false}
      onClick={[Function]}
      size="small"
      tabIndex={0}
      type="button"
    >
      <ForwardRef(Button)
        className="security--button"
        disabled={false}
        kind="ghost"
        onClick={[Function]}
        size="small"
        tabIndex={0}
        type="button"
      >
        <button
          className="security--button bx--btn bx--btn--sm bx--btn--ghost"
          disabled={false}
          onClick={[Function]}
          tabIndex={0}
          type="button"
        >
          Add
        </button>
      </ForwardRef(Button)>
    </Button>
  </section>
</TagWall>
`;

exports[`TagWall Rendering renders the HTML of the node's subtree 1`] = `
<section
  aria-label="Label"
  class="security--tag-wall"
>
  <h2
    class="security--tag-wall__label"
  >
    Label
  </h2>
  <span
    class="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--removable security--tag--interactive--selected"
  >
    Tag 1
    <button
      aria-label=""
      class="security--tag--interactive__button"
    >
      <svg
        aria-hidden="true"
        class="security--icon security--tag--interactive__button__icon"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        style="will-change: transform;"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 4.7L11.3 4 8 7.3 4.7 4 4 4.7 7.3 8 4 11.3 4.7 12 8 8.7 11.3 12 12 11.3 8.7 8z"
        />
      </svg>
    </button>
  </span>
  <span
    class="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--removable security--tag--interactive--selected"
  >
    Tag 2
    <button
      aria-label=""
      class="security--tag--interactive__button"
    >
      <svg
        aria-hidden="true"
        class="security--icon security--tag--interactive__button__icon"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        style="will-change: transform;"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 4.7L11.3 4 8 7.3 4.7 4 4 4.7 7.3 8 4 11.3 4.7 12 8 8.7 11.3 12 12 11.3 8.7 8z"
        />
      </svg>
    </button>
  </span>
  <span
    class="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--removable security--tag--interactive--selected"
  >
    Tag 3
    <button
      aria-label=""
      class="security--tag--interactive__button"
    >
      <svg
        aria-hidden="true"
        class="security--icon security--tag--interactive__button__icon"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        style="will-change: transform;"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 4.7L11.3 4 8 7.3 4.7 4 4 4.7 7.3 8 4 11.3 4.7 12 8 8.7 11.3 12 12 11.3 8.7 8z"
        />
      </svg>
    </button>
  </span>
  <span
    class="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--default security--tag--interactive--removable"
  >
    Tag 4
    <button
      aria-label=""
      class="security--tag--interactive__button"
    >
      <svg
        aria-hidden="true"
        class="security--icon security--tag--interactive__button__icon"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        style="will-change: transform;"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 4.7L11.3 4 8 7.3 4.7 4 4 4.7 7.3 8 4 11.3 4.7 12 8 8.7 11.3 12 12 11.3 8.7 8z"
        />
      </svg>
    </button>
  </span>
  <span
    class="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--default security--tag--interactive--removable"
  >
    Tag 5
    <button
      aria-label=""
      class="security--tag--interactive__button"
    >
      <svg
        aria-hidden="true"
        class="security--icon security--tag--interactive__button__icon"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        style="will-change: transform;"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 4.7L11.3 4 8 7.3 4.7 4 4 4.7 7.3 8 4 11.3 4.7 12 8 8.7 11.3 12 12 11.3 8.7 8z"
        />
      </svg>
    </button>
  </span>
  <span
    class="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--default security--tag--interactive--removable"
  >
    Tag 6
    <button
      aria-label=""
      class="security--tag--interactive__button"
    >
      <svg
        aria-hidden="true"
        class="security--icon security--tag--interactive__button__icon"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        style="will-change: transform;"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 4.7L11.3 4 8 7.3 4.7 4 4 4.7 7.3 8 4 11.3 4.7 12 8 8.7 11.3 12 12 11.3 8.7 8z"
        />
      </svg>
    </button>
  </span>
  <span
    class="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--default security--tag--interactive--removable"
  >
    Tag 7
    <button
      aria-label=""
      class="security--tag--interactive__button"
    >
      <svg
        aria-hidden="true"
        class="security--icon security--tag--interactive__button__icon"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        style="will-change: transform;"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 4.7L11.3 4 8 7.3 4.7 4 4 4.7 7.3 8 4 11.3 4.7 12 8 8.7 11.3 12 12 11.3 8.7 8z"
        />
      </svg>
    </button>
  </span>
  <span
    class="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--default security--tag--interactive--removable"
  >
    Tag 8
    <button
      aria-label=""
      class="security--tag--interactive__button"
    >
      <svg
        aria-hidden="true"
        class="security--icon security--tag--interactive__button__icon"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        style="will-change: transform;"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 4.7L11.3 4 8 7.3 4.7 4 4 4.7 7.3 8 4 11.3 4.7 12 8 8.7 11.3 12 12 11.3 8.7 8z"
        />
      </svg>
    </button>
  </span>
  <button
    class="security--button bx--btn bx--btn--sm bx--btn--ghost"
    tabindex="0"
    type="button"
  >
    Add
  </button>
</section>
`;

exports[`TagWall Rendering renders the disabled variation 1`] = `
<TagWall
  addButtonDisabled={false}
  addLabel="Add"
  className=""
  disable={true}
  itemToString={[Function]}
  items={
    Array [
      Object {
        "id": "0",
        "isSelected": true,
        "label": "Tag 1",
      },
      Object {
        "id": "1",
        "isSelected": true,
        "label": "Tag 2",
      },
      Object {
        "id": "2",
        "isSelected": true,
        "label": "Tag 3",
      },
      Object {
        "id": "3",
        "isSelected": false,
        "label": "Tag 4",
      },
      Object {
        "id": "4",
        "isSelected": false,
        "label": "Tag 5",
      },
      Object {
        "id": "5",
        "isSelected": false,
        "label": "Tag 6",
      },
      Object {
        "id": "6",
        "isSelected": false,
        "label": "Tag 7",
      },
      Object {
        "id": "7",
        "isSelected": false,
        "label": "Tag 8",
      },
    ]
  }
  label="Label"
  labels={Object {}}
  onAddButton={[Function]}
  onChange={[Function]}
>
  <section
    aria-label="Label"
    className="security--tag-wall"
  >
    <h2
      className="security--tag-wall__label"
    >
      Label
    </h2>
    <InteractiveTag
      isSelected={true}
      key="tag__0"
      onRemove={[Function]}
      removable={false}
      removeBtnLabel=""
      type="gray"
    >
      <Tag
        className="security--tag--interactive security--tag--interactive--selected"
        filter={false}
        type="gray"
      >
        <span
          className="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--selected"
        >
          Tag 1
        </span>
      </Tag>
    </InteractiveTag>
    <InteractiveTag
      isSelected={true}
      key="tag__1"
      onRemove={[Function]}
      removable={false}
      removeBtnLabel=""
      type="gray"
    >
      <Tag
        className="security--tag--interactive security--tag--interactive--selected"
        filter={false}
        type="gray"
      >
        <span
          className="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--selected"
        >
          Tag 2
        </span>
      </Tag>
    </InteractiveTag>
    <InteractiveTag
      isSelected={true}
      key="tag__2"
      onRemove={[Function]}
      removable={false}
      removeBtnLabel=""
      type="gray"
    >
      <Tag
        className="security--tag--interactive security--tag--interactive--selected"
        filter={false}
        type="gray"
      >
        <span
          className="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--selected"
        >
          Tag 3
        </span>
      </Tag>
    </InteractiveTag>
    <InteractiveTag
      isSelected={false}
      key="tag__3"
      onRemove={[Function]}
      removable={false}
      removeBtnLabel=""
      type="gray"
    >
      <Tag
        className="security--tag--interactive security--tag--interactive--default"
        filter={false}
        type="gray"
      >
        <span
          className="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--default"
        >
          Tag 4
        </span>
      </Tag>
    </InteractiveTag>
    <InteractiveTag
      isSelected={false}
      key="tag__4"
      onRemove={[Function]}
      removable={false}
      removeBtnLabel=""
      type="gray"
    >
      <Tag
        className="security--tag--interactive security--tag--interactive--default"
        filter={false}
        type="gray"
      >
        <span
          className="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--default"
        >
          Tag 5
        </span>
      </Tag>
    </InteractiveTag>
    <InteractiveTag
      isSelected={false}
      key="tag__5"
      onRemove={[Function]}
      removable={false}
      removeBtnLabel=""
      type="gray"
    >
      <Tag
        className="security--tag--interactive security--tag--interactive--default"
        filter={false}
        type="gray"
      >
        <span
          className="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--default"
        >
          Tag 6
        </span>
      </Tag>
    </InteractiveTag>
    <InteractiveTag
      isSelected={false}
      key="tag__6"
      onRemove={[Function]}
      removable={false}
      removeBtnLabel=""
      type="gray"
    >
      <Tag
        className="security--tag--interactive security--tag--interactive--default"
        filter={false}
        type="gray"
      >
        <span
          className="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--default"
        >
          Tag 7
        </span>
      </Tag>
    </InteractiveTag>
    <InteractiveTag
      isSelected={false}
      key="tag__7"
      onRemove={[Function]}
      removable={false}
      removeBtnLabel=""
      type="gray"
    >
      <Tag
        className="security--tag--interactive security--tag--interactive--default"
        filter={false}
        type="gray"
      >
        <span
          className="bx--tag bx--tag--gray security--tag--interactive security--tag--interactive--default"
        >
          Tag 8
        </span>
      </Tag>
    </InteractiveTag>
    <Button
      disabled={true}
      kind="ghost"
      largeText={null}
      loading={false}
      onClick={[Function]}
      size="small"
      tabIndex={0}
      type="button"
    >
      <ForwardRef(Button)
        className="security--button"
        disabled={true}
        kind="ghost"
        onClick={[Function]}
        size="small"
        tabIndex={0}
        type="button"
      >
        <button
          className="security--button bx--btn bx--btn--sm bx--btn--ghost bx--btn--disabled"
          disabled={true}
          onClick={[Function]}
          tabIndex={0}
          type="button"
        >
          Add
        </button>
      </ForwardRef(Button)>
    </Button>
  </section>
</TagWall>
`;
